packages/core/core/src/Strapi.ts

```diff
@@ -441,6 +441,11 @@ class Strapi extends Container implements Core.Strapi {
 
     await this.db.schema.sync();
 
+    // Remove orphaned components (entire types, not ids)
+    await this.db.repair({
```

Comment:
Just a quick 1st comment we should avoid as much as possible to run this at startup time. We have to reduce the startup work as much as possible :)

Main question is what happens if it doesn't run until an action triggers it ? (CTB CRUD/CLI cmd/...)

Reply:
I'll keep looking for a more specific point we could run this, since it does add one query for every content type that uses components on startup which would be good to avoid.

I'm not sure exactly when this error occurs, but I assume startup: https://github.com/strapi/strapi/issues/20931#issuecomment-2268589861

If I can pinpoint where that happens, maybe it could go in a try/catch and only run this repair when it fails.

Reply:
Looks like in this particular case it mostly related to CTB actions ?

I would consider that if someone manualy changes a schema they should run the function themselves (via CLI eventually) though

Reply:
Updated to only run on schema diff changes rather than every startup